theme/extend.go

```diff
@@ -0,0 +1,46 @@
+package theme
+
+import (
+	"image/color"
+
+	"fyne.io/fyne"
+)
+
+// ExtendDefaultTheme allows a theme to specify alterations to the default theme.
+// It will delegate all colour, theme or font lookups ignored to the built-in theme.
+func ExtendDefaultTheme(t fyne.Theme) fyne.Theme {
```

Comment:
What is the advantage of using

```go
func newMyTheme() fyne.Theme {
    return ExtendDefaultTheme(&myTheme{})
}

type myTheme struct {}

func (t *myTheme)Color(n fyne.ThemeColorName, v fyne.ThemeVariant) color.Color {
    if whatever {
        return color.Black
    }
    return nil
}
```

instead of

```go
func newMyTheme() fyne.Theme {
    return &myTheme{}
}

type myTheme struct {}

func (t *myTheme)Color(n fyne.ThemeColorName, v fyne.ThemeVariant) color.Color {
    if whatever {
        return color.Black
    }
    return theme.Current().Color(n, v)
}
```

The only thing latter would need is to give access to the current theme or define `Color()`, `Size()` and `Font()` on `theme` package.
I think the latter would be easier to understand.

Reply:
Okay, maybe we should export `theme.Default()` instead of creating a new type, then?

Reply:
Agreed. However, for me it feels weird to have to call a special method instead of simply wrapping up. Maybe we should not deprecate `theme.DefaultTheme()` but instead let it cache and create the _one_ default theme instance?

Reply:
I'm not sure which deprecation you are referring to, I can't find `theme.DefaultTheme()` in 1.4.
Can you expand on what you mean by "simply wrapping up" - I would have thought "ExtendDefaultTheme" was doing wrapping, so I can't tell the difference.

Reply:
Sorry, my fault. I misread it.
So, I suggest to let `theme.DefaultTheme()` cache the _one_ default theme instance.
Then one could simply do:
```go
func newMyTheme() fyne.Theme {
    return &myTheme{}
}

type myTheme struct {}

func (t *myTheme)Color(n fyne.ThemeColorName, v fyne.ThemeVariant) color.Color {
    if whatever {
        return color.Black
    }
    return theme.DefaultTheme().Color(n, v)
}
```

Reply:
Hmm. So you propose adding `DefaultTheme` in preference to `ExtendDefaultTheme` and require that all themes be complete through delegation rather than allowing partial implementations?

I don't feel particularly that one API is stronger than the other.

Reply:
I suggest that we use the mechanisms that are built into Go for embedding instead of implementing other patterns for the same goal.
My suggestion saves the whole `themeWrapper` and the `ExtendDefaultTheme()` method and does not require a single line of extra code.
It even does not _need_ the cache for the default theme but of course its performance would be bad if the default theme would be initialized on every call to `DefaultTheme()`. Even if you still disagree to my suggestion, the `DefaultTheme()` method should not create a new default theme every time.